source/val/validate_memory.cpp

```diff
@@ -785,7 +785,8 @@ spv_result_t ValidateVariable(ValidationState_t& _, const Instruction* inst) {
         (!_.HasCapability(spv::Capability::Float16) &&
          _.ContainsSizedIntOrFloatType(value_id, spv::Op::OpTypeFloat, 16))) {
       auto underlying_type = value_type;
-      while (underlying_type->opcode() == spv::Op::OpTypePointer) {
+      while (underlying_type &&
```

Comment:
This seems like a functional change, or at least not a clang-tidy change.  Is it already clear that value_type is not zero?
Should a new test be added for this?

I recommend pulling this part of the PR out to another. And at that point it's probably better to make value_type const and check it at its definition. at line 419 or so. That will save other guards in this function.

Reply:
Yeah, felt weird to patch this, but those were indeed found through the static analyzer.
I reverted those, and I agree, some refactoring might be better.